### ISSUE TITLE
Docker tests fail due to CSR dependency

### ISSUE THREAD

**Issue by eeaton** [terraform-google-modules/terraform-example-foundation#1309]
### TL;DR

For new contributors, it is impossible to follow the guidance at contributing.md and run `make docker_test_prepare` due to a dependency on CSR, which is deprecated.

This issue is seen in https://github.com/terraform-google-modules/terraform-example-foundation/pull/1304, and makes it much harder for a contributor to make a relatively simple PR contribution because they are unable to check CI tests themselves.

### Expected behavior

docker tests to create resources and run tests against them should not depend on CSR, because the docker tests are not actually creating any version-controlled repos. 

This should be available to any customer environment, regardless of whether they are grandfathered in to continue using CSR because they had enabled the API prior to deprecation.

### Observed behavior

docker tests fail, because there are [multiple places](https://github.com/search?q=repo%3Aterraform-google-modules%2Fterraform-example-foundation+sourcerepo.googleapis.com&type=code) that try to enable `sourcerepos.googleapis.com` and then run tests checking that `sourcerepos.googleapis.com` must be enabled. 

### Terraform Configuration

```hcl
See all instances where CSR is enabled or checked for in CI tests:
https://github.com/search?q=repo%3Aterraform-google-modules%2Fterraform-example-foundation+sourcerepo.googleapis.com&type=code
```

### Terraform Version

```sh
same outcome with any Terraform version.

The determining factor whether this works or fails is whether the customer Organization had used CSR prior to [deprecation on June 17 2024](https://cloud.google.com/source-repositories/docs?), in which case they have a limited legacy exception to continue using CSR. All new customer orgs will fail.
```
